### ISSUE TITLE
ability to cancel the slide change from start event handler

### ISSUE THREAD

**Issue by makc** [cmswalker/fullpage-react#79]
It would be helpful to maybe check return value [here](https://github.com/cmswalker/fullpage-react/blob/a607d397d4c245fc9e591a7e52a57a3248747854/lib/components/Fullpage.js#L313)

I am trying to work with short speed values and it scrolls past multiple slides, so I want to force some minimum time in the slide but there seems to be no way around it.

**Comment by makc** [comment]
so far hacked it to be this:
```javascript
  handleScroll(node, winProp, to, newState, compName, cb = noOp) {
    const ss = this.ss || ssStub();
    ss.flush();

const cancel =
    this.onSlideChangeStart(compName, this.props, this.state, newState[compName] || newState);

if (cancel && (winProp == 'scrollTop')) {
  node.scrollTop = this.state.activeSlide * window.innerHeight;
  this.setState({ scrollPending: false }, () => {
    console.log('slide change cancelled');
    setTimeout(() => {
      ss.flush();
      ss.listen();
    }, TIMEOUT);
  });
  return;
}

    scrollTo(node, winProp, to, this.scrollSpeed, () => {
```
and on the app side
```javascript
  onSlideChangeStart = (name, props, state, newState) => {
    let time = Date.now()
    if (time - this.lastSlideChangeTime < 1000) {
      return true
    }
    if (state.scrollPending) {
      this.lastSlideChangeTime = time
    }
```
seems to work, but I'd rather have official fix )

**Comment by cmswalker** [comment]
I see. 

We should be strict with it. We'll need to make sure both `onSlideChangeStart` and `onSlideChange` end return `truthy/falsy`. How about if `true` the scroll/slide remains locked and if `false/!!undefined` the scroll is allowed to go through?

**Comment by makc** [comment]
I thought returning true to block the change would be backwards-compatible, that's why.

**Comment by cmswalker** [comment]
Fixed in `3.0.30`
